### PR TITLE
Fixed #25378 -- Improved tab through experience in admin.

### PR DESCRIPTION
**Pull request by ryangallen**

### CODE REVIEW

File: django/contrib/admin/static/admin/css/widgets.css [hunk 1]
```diff
@@ -90,26 +90,32 @@
     display: block;
     text-indent: -3000px;
     overflow: hidden;
+    cursor: default;
+    opacity: 0.3;
```
Comment (ryangallen) [inline django/contrib/admin/static/admin/css/widgets.css]: This widget originally used color changes to indicate the button being enabled/disabled (using `.active` ...bad class name, might change that in another commit). This commit switches that indication to the more conventionally used opacity property.

The color change is now used for the :hover and :focus pseudo-classes, which I also believe is more conventional.

**Comment by ryangallen** [comment]
@elky can you review this PR?

**Comment by elky** [comment]
Do we really need color indication for `:focus` states? Most browsers do their own implementation like this:

screenshot 2015-09-12 at 02 37 31
<hr>
screenshot 2015-09-12 at 02 38 30
<hr>
screenshot 2015-09-12 at 02 38 44

This is on OSX. Windows does the same thing as fas as I know.

Is there any advantages of your solution?

File: django/contrib/admin/static/admin/css/base.css [hunk 2]
```diff
@@ -20,10 +20,15 @@ a:link, a:visited {
     text-decoration: none;
 }
 
-a:hover {
+a:focus, a:hover {
     color: #036;
 }
 
+a:focus {
+    outline: none;
```
Comment (elky) [inline django/contrib/admin/static/admin/css/base.css]: I don't think this is good in terms of accessibility and UX. It's fine to have our own `:focus` states but I think we should respect people's expectations -- users who primarily navigate by keyboard would expect to see the same (default) tab indication  as on other sites. 

Let's discuss it.

Comment (aaugustin) [inline django/contrib/admin/static/admin/css/base.css]: The admin has a discreet design. It aims at *not* carrying a distinct brand.

For this reason I agree that it's better to stick to the platform's conventions.

Comment (ryangallen) [inline django/contrib/admin/static/admin/css/base.css]: Thanks for the feedback guys. My originally thinking was not to rely on the browser and ensure the tabbed item is identifiable by explicitly setting it. Seems that most people agree it's best not to remove the default browser outline so I'll restore it where it was active. Some elements currently do not have any indication of being the focused item, even with the browser outline active, so I think it's best to keep those changes.

**Comment by ryangallen** [comment]
I removed the `outline: none;` settings so the default browser outline is left intact for all elements. Now there should be no unexpected behavior for tab users. I do think that setting the `:hover` styling for `:focus` as well compliments the tab through experience, and is a good fallback if a browser for some reason doesn't have the built-in blue outline.

**Comment by timgraham** [comment]
A quick test on Linux looks okay to me. Any more concerns?

**Comment by elky** [comment]
I'm going to do review code to make sure that all OK

**Comment by timgraham** [comment]
It's not critical, but if you have time to review it by Monday, we can include it in the 1.9 alpha release. Thanks!

File: django/contrib/admin/static/admin/css/widgets.css (hunk 1 above)
Comment (elky) [inline django/contrib/admin/static/admin/css/widgets.css]: Good point @ryangallen. But can you please disable hover/focus effect for inactive states of this widget? Currently if user move cursor over the inactive link it changes its color which looks not intuitive in terms of UX.

**Comment by elky** [comment]
Code looks good for me, just noticed one only UX thing -- @ryangallen please see my comment above.

**Comment by timgraham** [comment]
Ryan, do you have time to make that update?

**Comment by timgraham** [comment]
@elky Since Ryan hasn't responded in a couple days, if you have time to provide a patch for your comments that'd be great. (Or Ryan if you are there, just let us know.)

**Comment by ryangallen** [comment]
I'll check it out, sorry for the delay!

**Comment by ryangallen** [comment]
Should be good to go.

File: django/contrib/admin/static/admin/css/widgets.css (hunk 1 above)
Comment (ryangallen) [inline django/contrib/admin/static/admin/css/widgets.css]: Righto. https://github.com/ryangallen/django/commit/080cf1ef3fd8e80cb7938f35abfe0a4d95c0ea83 should fix that

**Comment by elky** [comment]
Cool. Thanks @ryangallen! @timgraham I think it's ready to merge if it's possible according to release plan.

**Comment by timgraham** [comment]
merged in 25cccca0a286585b9896546d57559e41b4e2ca16, thanks both!

**Pull request closed without merging**